configuration/azure/appconfig/appconfig.go

```diff
@@ -160,32 +160,71 @@ func parseMetadata(meta configuration.Metadata) (metadata, error) {
 func (r *ConfigurationStore) Get(ctx context.Context, req *configuration.GetRequest) (*configuration.GetResponse, error) {
 	keys := req.Keys
 	items := make([]*configuration.Item, 0, len(keys))
-	for _, key := range keys {
-		resp, err := r.client.GetSetting(
-			ctx,
-			key,
-			&azappconfig.GetSettingOptions{
-				Label: to.Ptr("label"),
-			})
-		if err != nil {
+
+	if len(keys) == 0 {
+		var err error
+		if items, err = r.getAll(ctx); err != nil {
 			return nil, err
 		}
-
-		item := &configuration.Item{
-			Metadata: map[string]string{},
+	} else {
+		for _, key := range keys {
+			resp, err := r.client.GetSetting(
+				ctx,
+				key,
+				&azappconfig.GetSettingOptions{
+					Label: to.Ptr("label"),
+				})
+			if err != nil {
+				return nil, err
+			}
+
+			item := &configuration.Item{
+				Metadata: map[string]string{},
+			}
+			item.Key = key
+			item.Value = *resp.Value
```

Comment:
Is there a chance that `resp.Value` and `resp.Label` could be nil? Should there be a nil-check before?

Reply:
Good one. If a value is not set API returns an empty string for Value but null for Label.